### PR TITLE
import keystore feature

### LINKED ISSUE #743: [Wallet] Import keystore
Add "keystore option" for import, user should be able to import (via file-picker or by drag/drop) an existing keystore

As part of #732

### PR DESCRIPTION
**Pull request by sarawut11**
closes #743

### CODE REVIEW

**Review by veado — CHANGES_REQUESTED** [review]

File: src/renderer/components/wallet/keystore/Keystore.styles.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+export const Form = styled(InnerForm)`
+  width: 100%;
+  padding: 30px;
+  padding-top: 15px;
+  text-align: -webkit-center;
+`
+
+export const KeystoreLabel = styled(A.Form.Item)`
+  font-family: Open Sans;
+  color: #4a5564;
```
[the remark this answers is not in the document]
Comment (sarawut11) [reply src/renderer/components/wallet/keystore/Keystore.styles.tsx:15]: I can't find same color in the themes. Can you share the file link you mentioned?

[the remark this answers is not in the document]
Comment (veado) [reply src/renderer/components/wallet/keystore/Keystore.styles.tsx:17]: Is this a color defined in [our Figma layouts](https://www.figma.com/file/LgRqZCGMShZ2mJ702MIqyV/ASGARDEX-2020)?

File: src/renderer/i18n/de/wallet.ts [hunk 2]
```diff
@@ -25,6 +25,9 @@ const wallet: WalletMessages = {
   'wallet.imports.phrase': 'Phrase',
   'wallet.imports.wallet': 'Importiere eine bestehende Wallet',
   'wallet.imports.enterphrase': 'Phrase eingeben',
+  'wallet.imports.error.instance': 'Could not create an instance of Client - DE',
+  'wallet.imports.error.keystore.load': 'Error while loading keystore - DE',
+  'wallet.imports.error.keystore.import': 'Error while importing keystore - DE',
```
Comment (veado) [inline src/renderer/i18n/de/wallet.ts:30]:
```
  'wallet.imports.error.instance': 'Es konnte keine Instanz vom Client erstellt werden',
  'wallet.imports.error.keystore.load': 'Fehler beim Laden der Keystore-Datei',
  'wallet.imports.error.keystore.import': 'Fehler beim Importieren der Keystore-Datei',
```

File: src/renderer/components/wallet/keystore/ImportKeystore.tsx [hunk 3]
```diff
@@ -1,78 +1,88 @@
...
   const intl = useIntl()
 
-  const { keystoreService } = useWalletContext()
-  const { loadKeystore } = keystoreService
-
   const { clientViewState$ } = useBinanceContext()
   const clientViewState = useObservableState(clientViewState$, 'notready')
   const [importing, setImporting] = useState(false)
-  const [importError, setImportError] = useState<Option<Error>>(none)
+  const [importError, setImportError] = useState<O.Option<Error>>(O.none)
```
Comment (veado) [inline src/renderer/components/wallet/keystore/ImportKeystore.tsx]: By using and subscribing to `importKeystore$` and `loadKeystore$`, we can store it's value (which is RemoteData) into a state. This state includes `pending`, `failure` and `success`. That's we can remove extra state handling of `importing`, `importError`.

To get an idea, check `Swap.tsx` component, where we deal with a value of `RemoteData` by subscribing to `swap$`. It's not the same stream or `RemoteData` here in ImportKeystore` we have to deal with, but the same idea behind.

**Review by veado — CHANGES_REQUESTED** [review]

**Review by veado — DISMISSED** [review]
:+1:

**Review by veado — APPROVED** [review]

File: src/renderer/components/wallet/keystore/Keystore.styles.tsx [hunk 4]
```diff
@@ -14,31 +13,38 @@ export const Form = styled(InnerForm)`
...
+//   border-radius: 4px;
+//   color: #50e3c2;
+//   font-size: 14px;
+//   cursor: pointer;
+//   display: flex;
+//   justify-content: center;
+//   align-items: center;
+//   margin-bottom: 50px;
+// `
+
```
Comment (veado) [inline src/renderer/components/wallet/keystore/Keystore.styles.tsx:47]: Do we still need these lines ^?

**Pull request merged**